packages/suite/src/components/suite/SecurityCheck/DeviceCompromised.tsx

```diff
@@ -3,14 +3,54 @@ import { Card } from '@trezor/components';
 import { TREZOR_SUPPORT_FW_REVISION_CHECK_FAILED_URL } from '@trezor/urls';
 
 import { WelcomeLayout } from 'src/components/suite';
-import { useDevice, useDispatch } from 'src/hooks/suite';
+import { useDevice, useDispatch, useSelector } from 'src/hooks/suite';
+import {
+    selectFirmwareHashCheckErrorIfEnabled,
+    selectFirmwareRevisionCheckErrorIfEnabled,
+} from 'src/reducers/suite/suiteReducer';
 
-import { SecurityCheckFail } from './SecurityCheckFail';
+import { SecurityCheckFail, SecurityCheckFailProps } from './SecurityCheckFail';
+import { hardFailureChecklistItems, softFailureChecklistItems } from './checklistItems';
+
+const useSecurityCheckFailProps = (): Partial<SecurityCheckFailProps> => {
+    const revisionCheckError = useSelector(selectFirmwareRevisionCheckErrorIfEnabled);
+    const hashCheckError = useSelector(selectFirmwareHashCheckErrorIfEnabled);
+
+    // revision check has precedence over hash check, because it is unimpeachable
```

Comment:
The comment could be a bit more descriptive, e.g.:

    // revision check has precedence over hash check, because it does not have the ambiguous other-error state